### PR TITLE
Verify the given interval and offset values of an updater are valid

### PR DESCRIPTION
**Pull request by nichamon**
The checks are
 * the given interval value is not an empty string,
 * the given interval value is a number,
 * the given interval value is greater than 0,
 * the given offset value is not an empty string,
 * the given offset value is a number.

`ldmsd_updtr_new_with_auth()` receives the interval and offset values as
char *. The checks are made before the function is called to keep the
patch small, i.e., no changes to the function signature and its
references. If the checks are made in the fucntion, there are no trivial
ways to send informative error messages to remote clients, e.g.,
`ldmsd_controller`.

### CODE REVIEW

**Comment by baallan** [comment]
I thought the interval == 0 was allowed for one-time immediate events.

If trapping errors later is hard, this location is also the place where we need a check for abs(offset ) <= interval/2 (or however we code the check to be consistent with the scheduling use of offset).

File: ldms/src/ldmsd/ldmsd_request.c [hunk 1]
```diff
@@ -2660,9 +2662,56 @@ static int updtr_add_handler(ldmsd_req_ctxt_t reqc)
...
+			cnt = Snprintf(&reqc->line_buf, &reqc->line_len,
+				"The given update interval value (%s) is not a number.",
+				interval_str);
+			goto send_reply;
+		} else {
+			if (0 >= interval) {
+				reqc->errcode = EINVAL;
+				cnt = Snprintf(&reqc->line_buf, &reqc->line_len,
+						"The update interval value must "
+						"not be or less than 0.");
```
Comment (tom95858) [inline ldms/src/ldmsd/ldmsd_request.c:2688]: @nichamon, please restate this as:
```
The update interval must be greater than zero
```

**Comment by tom95858** [comment]
@nichamon, also add argument checking in ldmsd_updtr_new_with_auth.

**Comment by tom95858** [comment]
@nichamon sorry one more thing, add an 'invalid argument' test(s) to the ldmsd_updtr_new_with_auth test cases.

File: ldms/src/ldmsd/ldmsd_request.c (hunk 1 above)
Comment (nichamon) [reply ldms/src/ldmsd/ldmsd_request.c]: Done

**Comment by nichamon** [comment]
> @nichamon, also add argument checking in ldmsd_updtr_new_with_auth.

Done

**Comment by nichamon** [comment]
> @nichamon sorry one more thing, add an 'invalid argument' test(s) to the ldmsd_updtr_new_with_auth test cases.

@tom95858 I'll add this to my list.

**Comment by nichamon** [comment]
> I thought the interval == 0 was allowed for one-time immediate events.
> 
> If trapping errors later is hard, this location is also the place where we need a check for abs(offset ) <= interval/2 (or however we code the check to be consistent with the scheduling use of offset).

@baallan I added the verification of offset. The logic used in this patch is the same as the one used to verify the sample offset.

**Comment by tom95858** [comment]
@nichamon, can you please add 
```
Fixes #423 
```
To the end of the commit message.

**Comment by nichamon** [comment]
> @nichamon, can you please add
> 
> ```
> Fixes #423 
> ```
> 
> To the end of the commit message.

@tom95858 Done.

**Pull request merged**